### PR TITLE
Spatial prior

### PR DESCRIPTION
**Pull request by PierreBoyeau**

### CODE REVIEW

File: simulations/spatial_prior/destvi_spatial/_model.py [hunk 1]
```diff
@@ -0,0 +1,413 @@
...
+from torch.utils.data import TensorDataset, DataLoader
+from torch.optim.lr_scheduler import ReduceLROnPlateau
+
+from scvi.data import register_tensor_from_anndata
+from scvi.model.base import BaseModelClass
+
+logger = logging.getLogger(__name__)
+
+
+class CustomTrainingPlan(TrainingPlan):
```
Comment (romain-lopez) [inline simulations/spatial_prior/destvi_spatial/_model.py:26]: to remove?

File: simulations/spatial_prior/destvi_spatial/_model.py [hunk 2]
```diff
@@ -0,0 +1,413 @@
...
+    def construct_loaders(
+        self,
+        use_gpu: Optional[bool] = None,
+        train_size: float = 0.9,
+        validation_size: Optional[float] = None,
+        batch_size: int = 128,
+        train_indices=None,
+        test_indices=None,
+    ):
+        from scvi import settings
```
Comment (romain-lopez) [inline simulations/spatial_prior/destvi_spatial/_model.py]: put import on top of file, run isort

File: simulations/spatial_prior/destvi_spatial/_model.py [hunk 3]
```diff
@@ -0,0 +1,413 @@
...
+        train_size: float = 0.9,
+        validation_size: Optional[float] = None,
+        batch_size: int = 128,
+        plan_kwargs: Optional[dict] = None,
+        plan_class: Optional[None] = None,
+        train_indices=None,
+        test_indices=None,
+        **kwargs,
+    ):
+        from scvi import settings
```
Comment (romain-lopez) [inline simulations/spatial_prior/destvi_spatial/_model.py]: remove

File: simulations/spatial_prior/destvi_spatial/_model.py [hunk 4]
```diff
@@ -0,0 +1,413 @@
...
+        except AttributeError:
+            self.history_ = None
+        self.module.eval()
+        if use_gpu:
+            self.module.cuda()
+        self.is_trained_ = True
+
+    @property
+    def _plan_class(self):
+        return CustomTrainingPlan
```
Comment (romain-lopez) [inline simulations/spatial_prior/destvi_spatial/_model.py:175]: not necessary

File: simulations/spatial_prior/destvi_spatial/_model.py [hunk 5]
```diff
@@ -0,0 +1,413 @@
...
+
+    @property
+    def _data_loader_cls(self):
+        return AnnDataLoader
+
+    def get_metric(self):
```
Comment (romain-lopez) [inline simulations/spatial_prior/destvi_spatial/_model.py:177]: @torch.nograd()?

**Review by romain-lopez — CHANGES_REQUESTED** [review]

**Comment by romain-lopez** [comment]
@PierreBoyeau, any news on this end btw?

**Comment by PierreBoyeau** [comment]
Hi @romain-lopez , I am harmonizing the code with the current codebase, so as to make my classes children of the updated classes. This should be done in a few hours!

Edit: Done!

**Review by romain-lopez — APPROVED** [review]

**Pull request merged**